pages/docs/deployment/self-host-v3.mdx

```diff
@@ -0,0 +1,350 @@
+You may use Valkey instead of Redis.
+
+### S3/Blob Store
+
+Langfuse requires an S3-compatible blob store to persist all incoming events, multi-modal inputs, and large exports.
+You can use a managed service on AWS, Azure, or GCP, or host it yourself using MinIO.
+
+### Clickhouse
+
+Langfuse uses Clickhouse as an OLAP database to store traces, observations, and scores.
+You can use the managed service by Clickhouse Cloud, or host it yourself.
+At least version 24.3 is required.
+
+## Deploying Langfuse
+
+Deploy the application container to your infrastructure.
+You can use managed services like AWS ECS, Azure Container Instances, or GCP Cloud Run, or host it yourself.
+
+During the container startup, all database migrations will be applied automatically.
+This can be optionally disabled via environment variables.
+
+### Run Langfuse Web
+
+```bash
+docker run --name langfuse \
+  -e DATABASE_URL=postgresql://hello \
+  -e NEXTAUTH_URL=http://localhost:3000 \
+  -e NEXTAUTH_SECRET=mysecret \
+  -e SALT=mysalt \
+  -e ENCRYPTION_KEY=0000000000000000000000000000000000000000000000000000000000000000 \ # generate via: openssl rand -hex 32
+  -e CLICKHOUSE_URL=http://clickhouse:8123 \
+  -e CLICKHOUSE_USER=clickhouse \
+  -e CLICKHOUSE_PASSWORD=clickhouse \
+  -e REDIS_HOST=localhost \
+  -e REDIS_PORT=6379 \
+  -e REDIS_AUTH=redis \
+  -e LANGFUSE_S3_EVENT_UPLOAD_ENABLED=true \
+  -e LANGFUSE_S3_EVENT_UPLOAD_BUCKET=my-bucket \
+  -e LANGFUSE_S3_EVENT_UPLOAD_REGION=us-east-1 \
+  -e LANGFUSE_S3_EVENT_UPLOAD_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE \
+  -e LANGFUSE_S3_EVENT_UPLOAD_SECRET_ACCESS_KEY=bPxRfiCYEXAMPLEKEY \
+  # Temp env variables during migration
+  -e LANGFUSE_ENABLE_EXPERIMENTAL_FEATURES=true \
+  -e LANGFUSE_ASYNC_INGESTION_PROCESSING=true \
+  -e LANGFUSE_ASYNC_CLICKHOUSE_INGESTION_PROCESSING=true \
+  -p 3000:3000 \
+  -a STDOUT \
+langfuse/langfuse:main # TODO: Update to tagged version
+```
+
+### Run Langfuse Worker
+
+```bash
+docker run --name langfuse \
+  -e DATABASE_URL=postgresql://hello \
+  -e SALT=mysalt \
+  -e ENCRYPTION_KEY=0000000000000000000000000000000000000000000000000000000000000000 \ # generate via: openssl rand -hex 32
+  -e CLICKHOUSE_URL=http://clickhouse:8123 \
+  -e CLICKHOUSE_USER=clickhouse \
+  -e CLICKHOUSE_PASSWORD=clickhouse \
+  -e REDIS_HOST=localhost \
+  -e REDIS_PORT=6379 \
+  -e REDIS_AUTH=redis \
+  -e LANGFUSE_S3_EVENT_UPLOAD_ENABLED=true \
+  -e LANGFUSE_S3_EVENT_UPLOAD_BUCKET=my-bucket \
+  -e LANGFUSE_S3_EVENT_UPLOAD_REGION=us-east-1 \
+  -e LANGFUSE_S3_EVENT_UPLOAD_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE \
+  -e LANGFUSE_S3_EVENT_UPLOAD_SECRET_ACCESS_KEY=bPxRfiCYEXAMPLEKEY \
+  # Temp env variables during migration
+  -e LANGFUSE_ENABLE_EXPERIMENTAL_FEATURES=true \
+  -e LANGFUSE_ASYNC_INGESTION_PROCESSING=true \
+  -e LANGFUSE_ASYNC_CLICKHOUSE_INGESTION_PROCESSING=true \
+  -p 3030:3030 \
+  -a STDOUT \
+langfuse/langfuse-worker:main # TODO: Update to tagged version
+```
+
+### Recommended sizing
```

Comment:
I'd probably create a new page for "Scaling langfuse" that helps with the configuration and includes details on how to scale databases (especially clickhouse)